cookbook/security/api_key_authentication.rst

```diff
@@ -67,7 +60,7 @@ value and then a User object is created::
         public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
         {
             $apiKey = $token->getCredentials();
-            $username = $this->userProvider->getUsernameForApiKey($apiKey);
+            $username = $userProvider->getUsernameForApiKey($apiKey);
```

Comment:
And we also have to ensure that the given `$userProvider` actually is an instance of `ApiKeyUserProvider`.